### DISCUSSION TITLE
** Breaking change starting August 18th **

### DISCUSSION THREAD

**Discussion by vadimazarov** [microsoft/Microsoft-Fabric-workload-development-sample#72] [Announcements]
### We are introducing several changes concerning the workload’s NuGet package: 

1.	**Updated limits for the package and file structure**:
We have updated the limits and restrictions of the NuGet package, including the number of items allowed, the expected folders, file size limit, file name limit and more. 
The complete list appears in this section [Microsoft Fabric workload environment (preview) - Microsoft Fabric | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Ffabric%2Fworkload-development-kit%2Fworkload-environment%23development-environment&data=05%7C02%7Cvadimazarov%40microsoft.com%7Cda1d33b35bd142fd041a08dcbf8642df%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638595830697178628%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=BIKdu8Ymu245pbDqHtwSNzNxRb2FzGTTR36%2F%2BjsF5g0%3D&reserved=0).
**Note** - we have updated the allowed icon types:

    -   The package can now contain **only** “.`jpeg`”, “.`jpg`”, and “.`png`”  icon types.
    
    -  `SVG` files are **no longer supported**. 

  This change is mandatory and will take effect starting **August 18th**. 

2.	**New developer package structure:** 
We have updated the ManifestPackageDebug.nuspec file ([https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/blob/main/Backend/src/Packages/manifest/ManifestPackageDebug.nuspec](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2FMicrosoft-Fabric-workload-development-sample%2Fblob%2Fmain%2FBackend%2Fsrc%2FPackages%2Fmanifest%2FManifestPackageDebug.nuspec&data=05%7C02%7Cvadimazarov%40microsoft.com%7Cda1d33b35bd142fd041a08dcbf8642df%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638595830697189585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=uH8SPoyjTfVVOy%2FAim4dtebmjFVsuGOD%2BirkwuCq0pU%3D&reserved=0)), and the developer package now contains the same resources as the release/test package. 
The purpose of this change is to allow early validation of the main scenarios for frontend resources during the development stage, though some differences may still exist. 
**Note – a valid package for developer scenario should contain BOTH backend and frontend resources.** 
This change is mandatory and will take effect starting **September 1st.** 
 
What you should do to prevent your workload from breaking:
1.	These changes apply to packages uploaded starting August 18th. _There is no effect on packages already uploaded_. 
2.	For newly uploaded packages, please read the documentation on limits and apply them to your package. An invalid package upload will result in an upload error, with the failure reason attached in the DevGateway logs or network traces.

**Comment by vishnu01** [comment]
where can we find DevGateway logs

**Reply by anitamayorov** [reply]
Hi @vishnu01 - can you clarify what logs are you looking for? The DevGateway itself has a pretty excessive logging of requests and errors if occurred.

**Comment by MikeCarlo** [comment]
What about *.ICO files

**Reply by anitamayorov** [reply]
Hi @MikeCarlo ,
.ico files are not supported, the allowed icon formats are `'.jpg'`, `'.jpeg'` and `'.png'`, see https://learn.microsoft.com/en-us/fabric/workload-development-kit/workload-environment#:~:text=Only%20the%20following%20file%20types%20are%20permitted%20in%20the%20Assets%20subfolder%3A%20.jpeg%2C%20.jpg%2C%20.png.
